src/Sylius/Behat/Context/Api/Admin/ManagingChannelsContext.php

```diff
@@ -0,0 +1,249 @@
+    }
+
+    /**
+     * @When I choose :locale as a default locale
+     */
+    public function iChooseAsADefaultLocale(LocaleInterface $locale): void
+    {
+        $this->client->addRequestData('defaultLocale', $this->iriConverter->getIriFromItem($locale));
+    }
+
+    /**
+     * @When I describe it as :description
+     */
+    public function iDescribeItAs(string $description): void
+    {
+        $this->client->addRequestData('description', $description);
+    }
+
+    /**
+     * @When I set its contact email as :contactEmail
+     */
+    public function iSetItsContactEmailAs(string $contactEmail): void
+    {
+        $this->client->addRequestData('contactEmail', $contactEmail);
+    }
+
+    /**
+     * @When I choose :country and :otherCountry as operating countries
+     */
+    public function iChooseAndAsOperatingCountries(CountryInterface $country, CountryInterface $otherCountry): void
+    {
+        $this->client->addRequestData('countries', [
+            $this->iriConverter->getIriFromItem($country),
+            $this->iriConverter->getIriFromItem($otherCountry),
+        ]);
+    }
+
+    /**
+     * @When I allow to skip shipping step if only one shipping method is available
+     */
+    public function iAllowToSkipShippingStepIfOnlyOneShippingMethodIsAvailable(): void
+    {
+        $this->client->addRequestData('skippingShippingStepAllowed', true);
+    }
+
+    /**
+     * @When I allow to skip payment step if only one payment method is available
+     */
+    public function iAllowToSkipPaymentStepIfOnlyOnePaymentMethodIsAvailable(): void
+    {
+        $this->client->addRequestData('skippingPaymentStepAllowed', true);
+    }
+
+    /**
+     * @When I specify menu taxon as :taxon
+     */
+    public function iSpecifyMenuTaxonAs(TaxonInterface $taxon): void
+    {
+        $this->client->addRequestData('menuTaxon', $this->iriConverter->getIriFromItem($taxon));
+    }
+
+    /**
+     * @When I specify company as :company
+     */
+    public function iSpecifyCompanyAs(string $company): void
+    {
+        $this->billingData['company'] = $company;
+    }
+
+    /**
+     * @When I specify tax ID as :taxId
+     */
+    public function iSpecifyTaxIdAs(string $taxId): void
+    {
+        $this->billingData['taxId'] = $taxId;
+    }
+
+    /**
+     * @When I specify shop billing address as :street, :postcode :city, :country
+     */
+    public function specifyShopBillingAddressAs(
+        string $street,
+        string $postcode,
+        string $city,
+        CountryInterface $country
+    ): void {
+        $this->billingData['street'] = $street;
+        $this->billingData['city'] = $city;
+        $this->billingData['postcode'] = $postcode;
+        $this->billingData['countryCode'] = $country->getCode();
+    }
+
+    /**
+     * @When I add it
+     */
+    public function iAddIt(): void
+    {
+        $this->createBillingData();
```

Comment:
It seems, that (as @GSadee proposed) we should allow working on shop billing data together with the channel. 

Something similar to https://symfonycasts.com/screencast/api-platform/embedded-write